File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> a = new HashSet<>();
        int max = 0;
        int l = 0;
        for(int r =0 ; r < s.length() ; r++){
            if(!a.contains(s.charAt(r))){
                a.add(s.charAt(r));
                max = Math.max(max , r - l + 1);
            }
            else{
                while(s.charAt(l) != s.charAt(r)){
                    a.remove(s.charAt(l));
                    l++;
                }
                a.remove(s.charAt(l));l++;
                a.add(s.charAt(r));
            }

        }
        return max;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<char> set = new HashSet<char>();
        int left = 0;
        int maxl = 0;
        for(int i = 0 ; i < s.Length ; i++){
            while(set.Contains(s[i])){
                set.Remove(s[left]);
                left++;
            }
            set.Add(s[i]);
            maxl = Math.Max(maxl , i - left + 1); 
        }
        return maxl;
    }
}